### ISSUE TITLE
primus-rooms crushes when joining room after re-connect.

### ISSUE THREAD

**Issue by bfunc** [cayasso/primus-rooms#68]
primus-rooms crushes when joining room after re-connect.

1. the issue happens is when clients re-connecting (local developent with webpack page reload on code change)
2. it is happens when more that one client (opened tab) are re-connecting at same time (one tab works well)

```node_modules\primus-rooms\lib\index.js:171
      return this._rooms[fn].apply(this._rooms, arguments);
                        ^
TypeError: Cannot read property 'join' of undefined 
                  at Sparky.<anonymous> (...\node_modules\primus-rooms\lib\index.js:171:25)
```

**Comment by kinsi55** [comment]
+1, Have you ever been able to figure out a fix for this?

**Comment by kinsi55** [comment]
Weird, hat it happen on the first day of using this in production today. I've changed the `spark.join()` to the `primus.join()` call now and check if `spark._rooms` exists beforehand. Also added a try/catch to the join call and just disconnect the spark if an error happens as a workaround for now.

**Comment by kinsi55** [comment]
Yep, spark._rooms is indeed pretty often not set in my case. I may just go ahead and build my own system as I only need a single room per client, might be the easiest fix for the while instead of digging trough this code of calls

**Comment by kinsi55** [comment]
Thought I'd go ahead and share what I did: I built my own minimal rooms plugin with focus on performance: https://github.com/kinsi55/node_primus-turborooms
